pkg/resource/out.go

```diff
@@ -67,10 +69,62 @@ func Out(request OutRequest, sourceDir string) (*OutResponse, error) {
 			return nil, fmt.Errorf("Couldn't create Container %s: %s", rsc.Container, err)
 		}
 	}
-	if _, err := client.ObjectPut(rsc.Container, filename, file, true, "", "", swift.Headers{}); err != nil {
-		return nil, fmt.Errorf("Failed to upload to swift: %s", err)
+
+	stat, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("Can't stats of file %s: %s", fileSource, err)
+	}
+
+	var bytes int64
+	bytes = stat.Size()
+	var fi os.FileInfo
+	if bytes > 134217728 {
```

Comment:
Is this size boundary arbitrarily chosen?

Reply:
It's proposed by a swift specialist ( larger 1GB )

Reply:
In stead of having a hidden magical value that triggers upload segmentation maybe the user should have more control when and if segmentation kicks in.